Formula/rbenv.rb

```diff
@@ -25,6 +26,8 @@ def install
       s.gsub! ":/usr/local/etc/rbenv.d", ":#{HOMEBREW_PREFIX}/etc/rbenv.d\\0" if HOMEBREW_PREFIX.to_s != "/usr/local"
     end
 
+    inreplace "libexec/rbenv-rehash", "$(command -v rbenv)", opt_bin/"rbenv"
```

Comment:
This works, but it feels brittle. If the expression `command -v rbenv` changes even slightly within rbenv source, this workaround silently breaks.

Reply:
There are two potential sources of breakage:
1. The [line](https://github.com/rbenv/rbenv/blob/585ed84283f3308380b843391ee7b12706ecff87/libexec/rbenv-rehash#L64) we want to change stops using `$(command -v rbenv)`.
2. `libexec/rbenv-rehash` starts using `$(command -v rbenv)` somewhere else we may not want to change.

Both seem sufficiently unlikely to me, but you'd know more about this than I do, of course. `libexec/rbenv-rehash` hasn't changed in years, and the relevant line we want to change has been there for even longer.

I'm not sure 2 is such a problem either, as I think I'm comfortable with using `opt_bin/"rbenv"` everywhere `rbenv` wants to do `command -v rbenv`.

1 should not be a problem as long as it doesn't happen simultaneously with 2, as the `inreplace` will error out when it is unable to do a replacement. We can avoid that by tightening the `inreplace` with more context from the line we actually want to replace.

Reply:
Thanks for elaborating. I'm not opposed to this being merged. But I do not feel like the fix is ideal in the long term. The main problem is that Homebrew aggressively prunes old versions of software after upgrades. This breaks not just rbenv, but also ruby gems linked to removed versions of software. I have fixed my own Ruby development environment with `export HOMEBREW_NO_INSTALL_CLEANUP=true`